docs/0002-atlas-translations-management.rst

```diff
@@ -0,0 +1,136 @@
+Atlas Translations Management Design
+####################################
+
+Date: 25 March 2024
+
+Status
+******
+Pending
+
+Context
+*******
+
+Open edX microservices and micro-frontends is using `OEP-58`_ to pull the latest
+translations from the `openedx-translations repository`_.
+
+The gist of `OEP-58`_ is two processes:
+
+- Extract the English translation source entries and commit them into the `openedx-translations repository`_
+- Provide a command to pull translations in both development and web server image builds
+
+Microservices, micro-frontends, XBlocks and plugins uses one or two files which makes it easier for Translators
+to locate and translate relevant resources.
+
+The `edx-platform`_ used to have 7 different resources, but this has been deprecated in favor of ``edx-platform.po``
+and ``edx-platform-js.po`` files. The details in the `edx-platform resources combination decision`_.
+
+The ``openedx-app-ios`` uses a clean architecture. Therefore, each app has its own i18n strings files. This is
+useful for Developer Experience, but we believe it's not good for Translator Experience.
+
+Decision
+********
+
+This decision will require few changes to the translation workflow in comparison with the old ``edx-app-ios``:
+
+- Translation files are no longer comitted directly in the mobie app ``openedx-app-ios/android`` repositories
+- Keep the i18n string files split into each clean architecture app without hurting Developer Experience.
+- Identical string IDs across the application will be treated as the same string/translation.
+  For example, ``BUTTON_CLOSE="Close"`` in dashboard is the same ``BUTTON_CLOSE="Close"`` in profile app.
+  Otherwise, a validation script will throw an error and prevent merging the pull request.
```

Comment:
Would it be possible to namespace these? So, for example, when compiling the strings into a resource file we could have `dashboard.BUTTON_CLOSE` and `profile.BUTTON_CLOSE`?

I'd think ideally this could happen as part of the source string compilation process, so each section could still just use `BUTTON_CLOSE`, but if that isn't feasible then directly adding the namespace to the strings in each section could work too.

Reply:
It's possible, but we'd rather keep the strings intact to avoid introducing unintended bugs.

@volodymyr-chekyrta what do you think?

Reply:
The same idea has already been suggested by Volodymyr. I'll mark this as resolved.

https://github.com/openedx/openedx-app-ios/pull/367#discussion_r1540988521

Thanks again!